pennylane/qchem/structure.py

```diff
@@ -326,7 +364,15 @@ def hf_state(electrons, orbitals):
 
     state = np.where(np.arange(orbitals) < electrons, 1, 0)
 
-    return np.array(state)
+    if basis == "parity":
+        pi_matrix = np.tril(np.ones((orbitals, orbitals)))
+        return (np.matmul(pi_matrix, state) % 2).astype("int32")
+
+    if basis == "bravyi_kitaev":
+        beta_matrix = _beta_matrix(orbitals)
+        return (np.matmul(beta_matrix, state) % 2).astype("int32")
```

Comment:
This should be `.astype(int)` for consistency with the occupation number `state`.